setup/helpers/abs_helper.rb

```diff
@@ -0,0 +1,657 @@
+  #
+  # @return [true,false] Based on whether the user has a valid token
+  #
+  # @example
+  #   success = abs_initialize
+  #
+  def abs_initialize
+    # only proceed if the user has a token
+    user_has_token = false
+    if get_abs_token
+      user_has_token = true
+
+      # TODO: preface all ABS environment variables with ABS?
+      @abs_base_url = ENV["ABS_BASE_URL"] ? ENV["ABS_BASE_URL"] : ABS_BASE_URL
+      @aws_platform = ENV["ABS_AWS_PLATFORM"] ? ENV["ABS_AWS_PLATFORM"] : AWS_PLATFORM
+      @aws_image_id = ENV["ABS_AWS_IMAGE_ID"] ? ENV["ABS_AWS_IMAGE_ID"] : AWS_IMAGE_ID
+      @aws_size = ENV["ABS_AWS_SIZE"] ? ENV["ABS_AWS_SIZE"] : AWS_SIZE
+      @aws_volume_size = ENV["ABS_AWS_VOLUME_SIZE"] ? ENV["ABS_AWS_VOLUME_SIZE"] : AWS_VOLUME_SIZE
+      @aws_region = ENV["ABS_AWS_REGION"] ? ENV["ABS_AWS_REGION"] : AWS_REGION
+      @aws_reap_time = ENV["ABS_AWS_REAP_TIME"] ? ENV["ABS_AWS_REAP_TIME"] : AWS_REAP_TIME
+      @mom_size = ENV["ABS_AWS_MOM_SIZE"] ? ENV["ABS_AWS_MOM_SIZE"] : MOM_SIZE
+      @mom_volume_size = ENV["ABS_AWS_MOM_VOLUME_SIZE"] ? ENV["ABS_AWS_MOM_VOLUME_SIZE"] : MOM_VOLUME_SIZE
+      @metrics_size = ENV["ABS_AWS_METRICS_SIZE"] ? ENV["ABS_AWS_METRICS_SIZE"] : METRICS_SIZE
+      @metrics_volume_size = ENV["ABS_METRICS_VOLUME_SIZE"] ? ENV["ABS_METRICS_VOLUME_SIZE"] : METRICS_VOLUME_SIZE
+      @abs_beaker_pe_version = ENV["BEAKER_PE_VER"] ? ENV["BEAKER_PE_VER"] : nil
+    end
+    user_has_token
+  end
+
+  # Initializes AbsHelper and returns the hosts for an ApplesToApples run
+  #
+  # @author Bill Claytor
+  #
+  # @return [Hash] The ApplesToApples hosts (mom and metrics)
+  #
+  # @example
+  #   hosts = abs_get_a2a_hosts
+  #
+  def get_a2a_hosts
+    abs_initialize
+    mom =
+      { 'role': "mom",
+        'size': @mom_size,
+        'volume_size': @mom_volume_size }
+
+    metrics =
+      { 'role': "metrics",
+        'size': @metrics_size,
+        'volume_size': @metrics_volume_size }
+
+    hosts = [mom, metrics]
+
+    return hosts
+  end
+
+  # Attempts to provision the specified hosts via ABS
+  #
+  # hosts will likely come from abs_get_a2a_hosts:
+  #  {"mom": @abs_aws_mom_size, "metrics": @abs_aws_metrics_size}
+  #
+  # otherwise specify hosts in the following format
+  #  {"role1": "size1", "role2": "size2", ... }
+  #
+  # @author Bill Claytor
+  #
+  # @param [Array] hosts_to_request The hosts to request via ABS
+  #
+  # @return [JSON, nil] The hosts that were provisioned via ABS, otherwise nil
+  #
+  # @example
+  #   abs_get_resource_hosts = abs_get_resource_hosts(hosts_to_request)
+  #
+  def get_abs_resource_hosts(hosts_to_request)
+    hosts = []
+    abs_resource_hosts = nil
+    invalid_response = false
+
+    puts
+    puts "Attempting to provision ABS hosts: #{hosts_to_request}"
+    puts
+
+    # ensure the user has a token before proceeding
+    if abs_initialize
+      uri = URI("#{@abs_base_url}/awsdirect")
+      hosts_to_request.each do |host_to_request|
```

Comment:
We should do this in parallel. Since in-parallel is a dependency for Beaker, you should just be able to:
require in_parallel

include InParallel

#options - 300 second timeout, and true to kill all threads on error in any of the threads
hosts_to_request.each.in_parallel(nil, 300, true) do |host_to_request|